client/src/components/desk/pickup/PhotoIdCheck.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import {Button, Card, Icon} from "semantic-ui-react";
+import {updateRequestStatus} from "../DeskUtil";
+import {FULFILLED} from "../../../types/Hardware";
+
+function generateCard({title, content, button, error}: any) {
+    return <Card style={{maxWidth: 200}}>
+        <Card.Content>
+            <strong>{title}</strong>
+        </Card.Content>
+        <Card.Content>
+            {content}
+        </Card.Content>
+        {error ? <Card.Content className="hw-negative">
+            <Icon name="warning sign"/>Unable to change request status: {error.message}
+        </Card.Content> : ""}
+        {button}
+    </Card>;
+}
+
+function PhotoIdCheck({userName, loading, updateRequest, requests, returnRequired, haveID, error}: any) {
+    const idRequired = generateCard({
+        title: <><Icon name="id badge"/>Photo ID required</>,
+        content: `Did you collect ${userName}'s photo ID?`,
+        button: <Button.Group>
+            <Button loading={loading} onClick={event =>
+                requests.forEach((request: any) =>
+                    updateRequestStatus(updateRequest, request.request_id, FULFILLED)
+                )
+            }>Skip</Button>
+            <Button color="green" loading={loading} onClick={event =>
+                requests.forEach((request: any) =>
+                    updateRequestStatus(updateRequest, request.request_id, FULFILLED, true)
+                )
+            }>Yes</Button>
+        </Button.Group>,
+        error
+    });
+
+    const noIdRequiredButton = <Button color="green" loading={loading} onClick={event =>
+        requests.forEach((request: any) =>
+            updateRequestStatus(updateRequest, request.request_id, FULFILLED)
+        )
+    }>Complete pickup</Button>;
+
+    const idAlreadyCollected = generateCard({
+        title: <><Icon name="check circle"/>All set!</>,
```

Comment:
this is wild... what exactly is this syntax

Reply:
If you saw it, it would make more sense.  When you click Fulfill, there's 3 possible Photo ID collection states:
1) No photo ID needs to be collected because it's not required
2) Photo ID needs to be collected
3) No photo ID needs to be collected because we already have it

I can simplify the function some though